pkg/pgmodel/migrate.go

```diff
@@ -6,48 +6,61 @@ package pgmodel
 
 import (
 	"bytes"
-	"database/sql"
+	"context"
 	"fmt"
 	"io"
-	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
 	"strings"
 	"sync"
 
-	"github.com/golang-migrate/migrate/v4"
-	"github.com/golang-migrate/migrate/v4/database/postgres"
-	"github.com/golang-migrate/migrate/v4/source"
-	"github.com/golang-migrate/migrate/v4/source/httpfs"
+	"github.com/blang/semver"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
 	"github.com/timescale/timescale-prometheus/pkg/log"
 	"github.com/timescale/timescale-prometheus/pkg/pgmodel/migrations"
 )
 
 const (
 	timescaleInstall            = "CREATE EXTENSION IF NOT EXISTS timescaledb WITH SCHEMA public;"
 	extensionInstall            = "CREATE EXTENSION IF NOT EXISTS timescale_prometheus_extra WITH SCHEMA %s;"
-	metadataUpdateWithExtension = "SELECT update_tsprom_metadata($1, $2, $3)"
+	metadataUpdateWithExtension = "SELECT %s.update_tsprom_metadata($1, $2, $3)"
 	metadataUpdateNoExtension   = "INSERT INTO _timescaledb_catalog.metadata(key, value, include_in_telemetry) VALUES ($1, $2, $3) ON CONFLICT (key) DO UPDATE SET value = EXCLUDED.value, include_in_telemetry = EXCLUDED.include_in_telemetry"
-)
+	createMigrationsTable       = "CREATE TABLE IF NOT EXISTS %s (version text not null primary key)"
+	getVersion                  = "SELECT version FROM %s LIMIT 1"
+	setVersion                  = "INSERT INTO %s (version) VALUES ($1)"
+	truncateMigrationsTable     = "TRUNCATE %s"
```

Comment:
if these aren't going to be user-settable then I think we should either do
```suggestion
	createMigrationsTable       = "CREATE TABLE IF NOT EXISTS prom_schema_migrations (version text not null primary key)"
	getVersion                  = "SELECT version FROM prom_schema_migrations LIMIT 1"
	setVersion                  = "INSERT INTO prom_schema_migrations (version) VALUES ($1)"
	truncateMigrationsTable     = "TRUNCATE prom_schema_migrations"
```
or at, if we must have the `migrationsTableName` constant, at least
```suggestion
	createMigrationsTable       = "CREATE TABLE IF NOT EXISTS "+ migrationsTableName + " (version text not null primary key)"
	getVersion                  = "SELECT version FROM "+ migrationsTableName + " LIMIT 1"
	setVersion                  = "INSERT INTO "+ migrationsTableName + " (version) VALUES ($1)"
	truncateMigrationsTable     = "TRUNCATE "+ migrationsTableName 
```
when reading through the control flow it's easier if you don't have to jump between the constants and `Sprintf` to figure out what's going on.